runelite-client/src/main/java/net/runelite/client/plugins/mta/enchantment/EnchantmentRoom.java

```diff
@@ -106,12 +105,15 @@ private WorldPoint findNearestStone()
 	@Subscribe
 	public void onItemSpawned(ItemSpawned itemSpawned)
 	{
-		final TileItem item = itemSpawned.getItem();
-		final Tile tile = itemSpawned.getTile();
+		if (!inside() || !config.enchantment())

```

Comment:
Generally the way we implement on/off options for features is by having the on/off toggle the overlay while having the code which does the actual tracking run regardless. This is so it allows you to toggle the option on/off and have that worked as expected instead of requiring a relog. This is breaking that.

Also I don't really see any need to modify the existing code.

Reply:
I've removed the config checks from here and `onItemDespawned()`. Although if there is a method to get all nearby items it may be better to repopulate the list of dragonstones when there's a config change.

Reply:
I would recommend just dropping this commit since it doesn't actually do anything and imo doesn't clean up anything either.

Reply:
Sure, the changes are pretty inconsequential.